crates/fluvio/src/stats/data_point.rs

```diff
@@ -0,0 +1,316 @@
+        if self.stats_collect() == ClientStatsDataCollect::None {
+            return Err(FluvioError::Other(
+                "Client is not collecting stats".to_string(),
+            ));
+        }
+
+        let mut header = vec![
+            "pid".to_string(),
+            "start_time_ns".to_string(),
+            "last_updated_ns".to_string(),
+        ];
+
+        let data_related = vec![
+            "offset".to_string(),
+            "total_records".to_string(),
+            "total_bytes".to_string(),
+            "last_latency_ns".to_string(),
+            "last_bytes".to_string(),
+            "throughput_kbps".to_string(),
+        ];
+
+        let system_related = vec!["last_mem_kb".to_string(), "last_cpu_pct".to_string()];
+
+        if (self.stats_collect() == ClientStatsDataCollect::All)
+            || (self.stats_collect() == ClientStatsDataCollect::Data)
+        {
+            header.extend(data_related);
+        }
+
+        if (self.stats_collect() == ClientStatsDataCollect::All)
+            || (self.stats_collect() == ClientStatsDataCollect::System)
+        {
+            header.extend(system_related);
+        }
+
+        // Render header as CSV w/ trailing newline
+        header.push("\n".to_string());
+        Ok(header.join(","))
+    }
+
+    /// Generates a CSV row. Columns returned based on client's configured `ClientStatsDataCollect`
+    pub fn csv_datapoint(&self) -> Result<String, FluvioError> {
+        if self.stats_collect() == ClientStatsDataCollect::None {
+            return Err(FluvioError::Other(
+                "Client is not collecting stats".to_string(),
+            ));
+        }
+
+        let mut data_point = vec![
+            self.pid().to_string(),
+            self.start_time().to_string(),
+            self.last_updated().to_string(),
+        ];
+
+        let data_related = vec![
+            self.offset().to_string(),
+            self.records().to_string(),
+            (self.total_bytes().equiv_amount(BYTE) as u64).to_string(),
+            self.last_latency().equiv_amount(NANOSECOND).to_string(),
+            (self.last_bytes().equiv_amount(BYTE) as u64).to_string(),
+            format!("{:.2}", self.throughput().equiv_amount(KILOBYTE_PER_SECOND)),
+        ];
+
+        let system_related = vec![
+            format!("{:.2}", self.memory().equiv_amount(KILOBYTE)),
+            self.cpu().to_string(),
+        ];
+
+        if (self.stats_collect() == ClientStatsDataCollect::All)
+            || (self.stats_collect() == ClientStatsDataCollect::Data)
+        {
+            data_point.extend(data_related);
+        }
+
+        if (self.stats_collect() == ClientStatsDataCollect::All)
+            || (self.stats_collect() == ClientStatsDataCollect::System)
+        {
+            data_point.extend(system_related);
+        }
+
+        // Render header as CSV w/ trailing newline
+        data_point.push("\n".to_string());
+        Ok(data_point.join(","))
```

Comment:
These two lines end up creating the data line with a trailing comma.  Appending the "\n" to the joined string would avoid that.